src/Bundle/Maker/Factory/FactoryGenerator.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Zenstruck\Foundry\Bundle\Maker\Factory;
+
+use Doctrine\Persistence\ManagerRegistry;
+use Symfony\Bundle\MakerBundle\Exception\RuntimeCommandException;
+use Symfony\Bundle\MakerBundle\Generator;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+/** @internal */
+final class FactoryGenerator
+{
+    /** @param \Traversable<int, DefaultPropertiesGuesser> $defaultPropertiesGuessers */
+    public function __construct(private ManagerRegistry $managerRegistry, private KernelInterface $kernel, private \Traversable $defaultPropertiesGuessers, private FactoryFinder $factoryFinder)
+    {
+    }
+
+    /**
+     * @param class-string $class
+     *
+     * @return class-string The factory's FQCN
+     */
+    public function generateFactory(string $class, MakeFactoryQuery $makeFactoryQuery): string
+    {
+        $generator = $makeFactoryQuery->getGenerator();
+
+        if (!\class_exists($class)) {
+            $class = $generator->createClassNameDetails($class, 'Entity\\')->getFullName();
+        }
+
+        if (!\class_exists($class)) {
+            throw new RuntimeCommandException(\sprintf('Class "%s" not found.', $makeFactoryQuery->getClass()));
+        }
+
+        $makeFactoryData = $this->createMakeFactoryData($class, $makeFactoryQuery->isPersisted());
+
+        $factory = $generator->createClassNameDetails(
+            $makeFactoryData->getObjectShortName(),
+            $this->guessNamespace($generator, $makeFactoryQuery->getNamespace(), $makeFactoryQuery->isTest()),
+            'Factory'
+        );
+
+        /** @var class-string $factoryClass */
+        $factoryClass = $factory->getFullName();
+
+        if (!$this->factoryFinder->classHasFactory($class)) {
+            $this->factoryFinder->addFactoryForClass($factoryClass, $class);
+        }
```

Comment:
_Registry_? I think I may have suggested this before and yeah, not really a registry...

Reply:
haha yes you've already suggested, but I don't want to call it _Registry_ because I think it would be confusing

if you don't think it's strange, I'm gonna go with `FactoryWithClassMap` or `FactoryClassMap`

Reply:
Yep FactoryClassMap sounds fine

Reply:
already removed :sweat_smile: